cmd/crank/beta/convert/pipelinecomposition/converter.go

```diff
@@ -147,15 +147,123 @@ func processFunctionInput(input *Input) *runtime.RawExtension {
 		"apiVersion":  "pt.fn.crossplane.io/v1beta1",
 		"kind":        "Resources",
 		"environment": processedInput.Environment.DeepCopy(),
-		"patchSets":   processedInput.PatchSets,
-		"resources":   processedInput.Resources,
+		"patchSets":   MigratePatchPolicy(processedInput.PatchSets),
+		"resources":   MigratePatchPolicyInResources(processedInput.Resources),
 	}
 
 	return &runtime.RawExtension{
 		Object: &unstructured.Unstructured{Object: inputType},
 	}
 }
 
+// ComposedTemplate composed template.
+type ComposedTemplate struct {
+	v1.ComposedTemplate
+
+	Patches []Patch `json:"patches,omitempty"`
+}
+
+// MigratePatchPolicyInResources patches resources.
+func MigratePatchPolicyInResources(resources []v1.ComposedTemplate) []ComposedTemplate {
+	composedTemplates := []ComposedTemplate{}
+
+	for _, resource := range resources {
+		composedTemplate := ComposedTemplate{}
+		composedTemplate.ComposedTemplate = resource
+
+		for _, patch := range resource.Patches {
+			newpatch := Patch{}
+			newpatch.Patch = patch
+
+			if patch.Policy != nil && patch.Policy.MergeOptions != nil {
+				newpatch.Policy = patchPolicy(patch.Policy)
+				newpatch.Patch.Policy = nil
+			}
+			composedTemplate.Patches = append(composedTemplate.Patches, newpatch)
+		}
+		composedTemplate.ComposedTemplate.Patches = nil
+		composedTemplates = append(composedTemplates, composedTemplate)
+	}
+	return composedTemplates
+}
+
+// MigratePatchPolicy migrates mergeoptions.
+func MigratePatchPolicy(patchset []v1.PatchSet) []NewPatchSet {
+	newPatchSets := []NewPatchSet{}
+
+	for _, patchSet := range patchset {
+		newpatchset := NewPatchSet{}
+		newpatchset.Name = patchSet.Name
+
+		for _, patch := range patchSet.Patches {
+			newpatch := Patch{}
+			newpatch.Patch = patch
+
+			if patch.Policy != nil && patch.Policy.MergeOptions != nil {
+				newpatch.Policy = patchPolicy(patch.Policy)
+				newpatch.Patch.Policy = nil
+			}
+
+			newpatchset.Patch = append(newpatchset.Patch, newpatch)
+		}
+
+		newPatchSets = append(newPatchSets, newpatchset)
+	}
+
+	return newPatchSets
+}
+
+func patchPolicy(policy *v1.PatchPolicy) *PatchPolicy {
+	if policy == nil {
+		return nil
+	}
+	mergeOptions := policy.MergeOptions
+	toFieldPath := ptr.To(ToFieldPathPolicyReplace)
+	if mergeOptions != nil {
+		if mergeOptions.KeepMapValues != nil && *mergeOptions.KeepMapValues {
+			toFieldPath = ptr.To(ToFieldPathPolicyMerge)
+		}
+		if mergeOptions.AppendSlice != nil && *mergeOptions.AppendSlice {
+			toFieldPath = ptr.To(ToFieldPathPolicyAppendArray)
+		}
+	}
+	return &PatchPolicy{
+		FromFieldPath: policy.FromFieldPath,
+		ToFieldPath:   toFieldPath,
+	}
+}
+
+// NewPatchSet test.
+type NewPatchSet struct {
```

Comment:
Let's move these types into [`types.go`](https://github.com/crossplane/crossplane/blob/master/cmd/crank/beta/convert/pipelinecomposition/types.go) so they are all discoverable in one place.

Reply:
Moved to types.go